segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/stat/AutoScaleProcessor.java

```diff
@@ -119,6 +119,7 @@ private void bootstrapRequestWriters() {
                                     ClientConfig.builder().controllerURI(configuration.getControllerUri())
                                                 .credentials(new DefaultCredentials(configuration.getAuthPassword(), configuration.getAuthUsername()))
                                                 .trustStore(configuration.getTlsCertFile())
+                                                .validateHostName(false)
```

Comment:
To my knowledge the only legit scenario for disabling hostname verification is when the truststore is specially crafted to allow ONLY the server's TLS certificate.   Please be extra careful about this.

Reply:
Will take care of this in #2582.

Reply:
This class doesnt make use of the REST interface to communicate with controller. 
why is this change relevant here?

Reply:
This is a temp fix for #2582. Will merge that PR into this before merging.